src/main/groovy/com/morpheusdata/hyperv/HyperVCloudProvider.groovy

```diff
@@ -768,21 +826,23 @@ class HyperVCloudProvider implements CloudProvider {
 	}
 
 	private resolveUniqueIdsToVMids(Map opts, node) {
+		log.debug("opts: ${opts}")
 		DataQuery query = new DatasetQuery().withFilters(
-				new DataFilter("zone", opts.zone),
+				new DataFilter("zone.id", opts.zone?.id),
```

Comment:
this is not going to return the correct results if zone.zone is null. What is the expected behavior if zone ID is 1 vs if zone is null?

Reply:
null check added for zone before query

Reply:
This whole method looks like it belongs in the virtual machine sync service, so we'll leave this for now and resolve this in the cache virtual machine story.